src/GafferUIModule/PathListingWidgetBinding.cpp

```diff
@@ -1761,20 +1794,38 @@ IECore::PathMatcher getExpansion( uint64_t treeViewAddress )
 	return model ? model->getExpansion() : IECore::PathMatcher();
 }
 
-void setSelection( uint64_t treeViewAddress, const IECore::PathMatcher &paths, bool scrollToFirst, bool expandNonLeaf )
+void setSelection( uint64_t treeViewAddress, object paths, bool scrollToFirst, bool expandNonLeaf )
 {
 	IECorePython::ScopedGILRelease gilRelease;
 
 	QTreeView *treeView = reinterpret_cast<QTreeView *>( treeViewAddress );
 	PathModel *model = dynamic_cast<PathModel *>( treeView->model() );
-	model->setSelection( paths, scrollToFirst, expandNonLeaf );
+
+	list pythonPaths = extract<list>( paths );
+	PathModel::Selection selection;
+	for( ssize_t i = 0, e = boost::python::len( pythonPaths ); i < e; ++i )
+	{
+		const IECore::PathMatcher &p = extract<const IECore::PathMatcher &>( pythonPaths[i] );
+		selection.push_back( p );
+	}
```

Comment:
This is using the Python API, so needs to be done outside of the ScopedGILRelease - I wouldn't be surprised if this is responsible for the crashes seen on CI. I also wouldn't be surprised if we could use `container_utils::extend_container` successfully once it's done outside the GIL release.